python/cudf/cudf/core/dataframe.py

```diff
@@ -783,11 +783,23 @@ def fallback(col, fn):
                 else:
                     result[col] = fallback(rhs._cols[col], _reverse_op(fn))
         elif isinstance(other, Series):
-            raise NotImplementedError(
-                "Series to DataFrame arithmetic not supported "
-                "until strings can be used as indices. Try converting your"
-                " Series into a DataFrame first."
-            )
+            result_cols = list(self.columns)
+            for new_col in list(other.index):
+                if new_col not in result_cols:
+                    result_cols.append(new_col)
+            for col in result_cols:
+                if col in self.columns and col in other.index:
+                    l_opr = self._cols[col]
+                    r_opr = other[other.index == col][0]
+                else:
+                    if col not in self.columns:
+                        l_opr = Series([np.nan] * len(self))
+                        r_opr = other[other.index == col][0]
+                    if col not in other.index:
+                        l_opr = self._cols[col]
+                        r_opr = np.nan
```

Comment:
```suggestion
                        r_opr = None
```

Reply:
Turns out this goes way down the rabbit hole of us not handling scalar `None` nicely in binaryops. I'm not sure of a good solution to this right now but we should be able to eventually construct a typed scalar with value of `null` in Cython to feed to a `vector_scalar` binaryop. Not sure if it's feasible for this PR though...

Reply:
For now using `nan` produces pretty close to the desired behavior in terms of arithmetic with the exception of the fact that `int` types plus `nan` gives us `float`. However thanks to the way pandas handles this, that's actually what pandas gets too, so it might be ok to leave as-is for now.

Reply:
opening an issue and a PR for this.